filament/include/filament/RenderableManager.h

```diff
@@ -463,6 +465,20 @@ class UTILS_PUBLIC RenderableManager : public FilamentAPI {
      */
     void setMorphWeights(Instance instance, math::float4 const& weights) noexcept;
 
+    /**
+     * Updates the morphing weights for the given primitive.
+     */
+    void setMorphWeightsAt(Instance instance,
+            size_t primitiveIndex, const float* weights, int count) noexcept;
+
+    /**
+     * Updates the morphing targets for the given primitive.
+     *
+     * Texture sampler must be SAMPLER_2D_ARRAY and the array size must be greater than weight count.
+     */
+    void setMorphTargetsAt(Instance instance,
```

Comment:
The fact that a texture is used should be an implementation detail and not exposed to the user. Also why a texture array? You don't seem to use slices anywhere?

Reply:
Because gltfio can only see public headers in filament and morph targets are created by gltfio. Thus I need a way setting morph targets texture to the primitive of renderable.

The number of morph targets could be a lot. In my case it's 52. If we create a texture per a morph target, we can't binding all samplers because the maximum sampler binding count is 16. So each primitive has a sampler array and each target are uploaded into each slice(even index is for position and odd index is for normal).

Reply:
The public API should receive morph weights as a float array and handle textures internally.

Reply:
Ok I will do that internally like SkinningBuffer. I think this class is very similar to this case.

From OpenGL ES 3.0 Spec, the minimum number of layers allowed in an array texture is at least 256. Thus if we make a limitation of Filament then it will be fine using texture array.